deployments/ansible/roles/collector/tasks/otel_win_reg.yml

```diff
@@ -1,17 +1,57 @@
 ---
-- name: Get Splunk OpenTelemetry Collector options list
+- name: Get Splunk OpenTelemetry Collector options list (including default options)
+  when: splunk_collector_msi_is_configurable is not defined
   set_fact:
     splunk_otel_collector_options_list: |-
       {%- set value = item.value -%}
       {{ (splunk_otel_collector_options_list | default([])) + [item.key + '=' + (value | string)] }}
   loop: >
     {{ splunk_otel_collector_options | default({}) | combine(splunk_otel_collector_additional_env_vars) | dict2items }}
 
+- name: Get Splunk OpenTelemetry Collector options list (only custom config variables
+  when: splunk_collector_msi_is_configurable is defined
+  set_fact:
+    splunk_otel_collector_options_list: |-
+      {%- set value = item.value -%}
+      {{ (splunk_otel_collector_options_list | default([])) + [item.key + '=' + (value | string)] }}
+  loop: >
+    {{ splunk_otel_collector_additional_env_vars | dict2items }}
+
+- name: Get existing registry contents
+  when:
+    - splunk_collector_msi_is_configurable is defined
+    - splunk_otel_collector_options_list is defined
+  ansible.windows.win_reg_stat:
+    path: "{{ splunk_otel_collector_service_registry_key }}"
+  register: registry_contents
+
+- debug:
+    var: registry_contents
+  when:
+    - splunk_collector_msi_is_configurable is defined
+    - splunk_otel_collector_options_list is defined
+
+- debug:
+    var: registry_contents.properties.Environment.value
+  when:
+    - splunk_collector_msi_is_configurable is defined
+    - splunk_otel_collector_options_list is defined
+
+- name: Append custom variables to existing registry contents
+  when:
+    - splunk_collector_msi_is_configurable is defined
+    - splunk_otel_collector_options_list is defined
+  set_fact:
+    splunk_otel_collector_options_list:
+      "{{ registry_contents.properties.Environment.value + splunk_otel_collector_options_list }}"
+
 - name: Set Splunk OpenTelemetry Collector registry value
+  when:
+    - splunk_otel_collector_options_list is defined
   ansible.windows.win_regedit:
     path: "{{ splunk_otel_collector_service_registry_key }}"
     state: present
     name: Environment
-    data: "{{ splunk_otel_collector_options_list | sort }}"
+    data: "{{ splunk_otel_collector_options_list | sort | unique }}"
```

Comment:
For the sake of [Ansible's idempotency check](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_intro.html#desired-state-and-idempotency), we need to make sure that entries in the registry are unique (which is why the `| unique` statement is added). Without this check, the custom variables are simply added again, resulting in duplicates.